src/foundry/client/core/issues.d.mts

```diff
@@ -0,0 +1,196 @@
+
+  /**
+   * A class responsible for tracking issues in the current world.
+   */
+  class ClientIssues {
+    /**
+     * Keep track of valid Documents in the world that are using module-provided sub-types.
+     * @remark the keys are module ids.
+     */
+    #moduleTypeMap: Map<string, ModuleSubTypeCounts>;
+
+    /**
+     * Keep track of document validation failures.
+     */
+    #documentValidationFailures: Record<
+      DocumentType,
+      Record<string, { name: string; error: DataModelValidationError }>
+    >;
+
+    /**
+     * Keep track of any usability issues related to browser or technology versions.
+     */
+    #usabilityIssues: Record<string, ClientIssues.UsabilityIssue>;
+
+    /** The minimum supported resolution. */
+    static #MIN_RESOLUTION: { WIDTH: 1024; HEIGHT: 700 };
+
+    /**
+     * The minimum supported client versions.
+     */
+    static #BROWSER_TESTS: {
+      /**
+       * @defaultValue
+       * ```typescript
+       * {
+       *   minimum: 24;
+       *   match: RegExp;
+       *   message: "ERROR.ElectronVersion";
+       * }
+       * ```
+       */
+      Electron: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 92;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Chromium: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 90;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Firefox: ClientIssues.BROWSER_TEST;
+      /**
+       * @defaultValue
+       * ```
+       * {
+       *   minimum: 15.4;
+       *   match: RegExp;
+       *   message: "ERROR.BrowserVersion";
+       * }
+       * ```
+       */
+      Safari: ClientIssues.BROWSER_TEST;
+    };
+
+    /**
+     * Add a Document to the count of module-provided sub-types.
+     * @param documentName      - The Document name.
+     * @param subType           - The Document's sub-type.
+     * @param options.decrement - Decrement the count rather than incrementing it. default: `false`
+     */
+    #countDocumentSubType(documentName: DocumentType, subType: string, options?: { decrement?: boolean }): void;
+
+    /**
+     * Detect the user's browser and display a notification if it is below the minimum required version.
+     */
+    #detectBrowserVersion(): void;
+
+    /**
+     * Record a reference to a resolution notification ID so that we can remove it if the problem is remedied.
+     */
+    #resolutionTooLowNotification: number;
+
+    /**
+     * Detect the user's resolution and display a notification if it is too small.
+     */
+    #detectResolution(): void;
+
+    /**
+     * Detect and display warnings for known performance issues which may occur due to the user's hardware or browser
+     * configuration.
+     * @internal
+     */
+    _detectWebGLIssues(): void;
```

Comment:
mark as protected and remove the internal tag

Reply:
Used in board.js.